subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -889,20 +859,283 @@ static int nrf_cloud_encode_service_info_ui(const struct nrf_cloud_svc_info_ui *
 	return 0;
 }
 
+#ifdef CONFIG_MODEM_INFO
+
+#define MODEM_INFO_JSON_KEY_NET_INF	"networkInfo"
+#define MODEM_INFO_JSON_KEY_SIM_INF	"simInfo"
+#define MODEM_INFO_JSON_KEY_DEV_INF	"deviceInfo"
+
+static int add_modem_info_data(struct lte_param *param, cJSON *json_obj)
+{
+	char data_name[MODEM_INFO_MAX_RESPONSE_SIZE];
+	enum at_param_type data_type;
+	int ret;
+
+	__ASSERT_NO_MSG(param != NULL);
+	__ASSERT_NO_MSG(json_obj != NULL);
+
+	memset(data_name, 0, ARRAY_SIZE(data_name));
+	ret = modem_info_name_get(param->type,
+				data_name);
+	if (ret < 0) {
+		LOG_DBG("Data name not obtained: %d", ret);
+		return -EINVAL;
+	}
+
+	data_type = modem_info_type_get(param->type);
+	if (data_type < 0) {
+		return -EINVAL;
+	}
+
+	if (data_type == AT_PARAM_TYPE_STRING &&
+	    param->type != MODEM_INFO_AREA_CODE) {
+		if (cJSON_AddStringToObject(json_obj, data_name, param->value_string) == NULL) {
+			return -ENOMEM;
+		}
+	} else {
+		if (cJSON_AddNumberToObject(json_obj, data_name, param->value) == NULL) {
+			return -ENOMEM;
+		}
+	}
+
+	return ret;
+}
+
+static int encode_modem_info_network(struct network_param *network, cJSON *json_obj)
+{
+	char network_mode[12] = {0};
+	char data_name[MODEM_INFO_MAX_RESPONSE_SIZE] = {0};
+	int ret;
+
+	__ASSERT_NO_MSG(network != NULL);
+	__ASSERT_NO_MSG(json_obj != NULL);
+
+	ret = add_modem_info_data(&network->current_band, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = add_modem_info_data(&network->sup_band, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = add_modem_info_data(&network->area_code, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = add_modem_info_data(&network->current_operator, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = add_modem_info_data(&network->ip_address, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = add_modem_info_data(&network->ue_mode, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	ret = modem_info_name_get(network->cellid_hex.type, data_name);
+	if (ret < 0) {
+		return ret;
+	}
+
+	if (cJSON_AddNumberToObjectCS(json_obj, data_name, network->cellid_dec) == NULL) {
+		return -EINVAL;
+	}
+
+	if (network->lte_mode.value == 1) {
+		strcat(network_mode, "LTE-M");
+	} else if (network->nbiot_mode.value == 1) {
+		strcat(network_mode, "NB-IoT");
+	}
+	if (network->gps_mode.value == 1) {
+		strcat(network_mode, " GPS");
+	}
+
+	if (cJSON_AddStringToObject(json_obj, "networkMode", network_mode) == NULL) {
+		return -EINVAL;
+	}
+
+	return ret;
+}
+
+static int encode_modem_info_sim(struct sim_param *sim, cJSON *json_obj)
+{
+	int ret;
+
+	__ASSERT_NO_MSG(sim != NULL);
+	__ASSERT_NO_MSG(json_obj != NULL);
+
+	ret = add_modem_info_data(&sim->uicc, json_obj);
+	if (ret) {
+		return ret;
+	}
+
+	if (IS_ENABLED(CONFIG_MODEM_INFO_ADD_SIM_ICCID)) {
```

Comment:
It seems odd to me that a KConfig which solely exists to configure the behavior of the codec is defined in the modem_info library

I think this KConfig option, and options like it, should be moved to the cloud_codec kconfig, and the naming changed to match.

Reply:
You're right. I'll go the other way and remove the dependency to those options in lib nrf_cloud. Now they're only used to determine if the ICCID/IMSI are added or not. If they are missing, `encode_modem_info_sim` will handle that as a non-critical error.